projects/vdk-heartbeat/src/vdk/internal/heartbeat/job_controller.py

```diff
@@ -206,6 +206,12 @@ def check_deployments(self, enabled=True, timeout_seconds=600):
             if not deployments:
                 log.info("No deployments so far. Will wait 30 seconds and try again.")
                 time.sleep(30)
+            elif deployments[0]["enabled"] != enabled:
```

Comment:
it's better try eveyr few seconds and timeout on 30 seconds instead of just sleeping

Reply:
30 seconds may not be enough, so setting a timeout at it is something I'd rather avoid.
Thank you for the review

Reply:
Well, the timeout must be configurable of course. 

The point is that having a fixed delay is not a good pattern because hard-coding a fixed timeout in tests can lead to inefficiencies and false negatives

 Even if the deployment happens in 1 second you still wait 30 seconds. 
If the deployment happens 35 seconds you are not flexible enough to cover it and the test would fail unnecessarily